### ISSUE TITLE
input of star aligner

### ISSUE THREAD

**Issue by Hecate08** [adamewing/bamsurgeon#61]
Hello,
We are using the star aligner for RNAseq reads. Now we wanted to validate our methods of variant calling by spiking variants using bamsurgeon. But .bam files from the star aligner are not submitted. Can you input the star aligner? That would be very helpful.
Thanks

**Comment by adamewing** [comment]
Sorry for the delay, STAR is now included as an alignment option. You'll need to specify ```--aligner STAR --alignopts STARrefdir:/path/to/STAR_reference_dir```

**Comment by Hecate08** [comment]
Thank you for this. 
I get now this error when I use STAR and 12 threads:
************************************************************
ERROR   2016-07-11 13:12:03.190345      encountered error in mutation spikein: ['chr10_6157266_6157266_0.94_None']
Traceback (most recent call last):
  File "/programs/Bamsurgeon/bamsurgeon/bin/addsnv.py", line 275, in makemut
    aligners.remap_bam(args.aligner, tmpoutbamname, args.refFasta, alignopts, mutid=hapstr, paired=(not args.single), picardjar=args.picardjar)
  File "build/bdist.linux-x86_64/egg/bamsurgeon/aligners.py", line 71, in remap_bam
    remap_STAR_bam(bamfn, threads, fastaref, picardjar, options['STARrefdir'], mutid=mutid, paired=paired)
  File "build/bdist.linux-x86_64/egg/bamsurgeon/aligners.py", line 410, in remap_STAR_bam
    raise ValueError("ERROR\t" + now() + "\t" + mutid + "\tbam readcount < fastq readcount, alignment sanity check failed!\n")
ValueError: ERROR       2016-07-11 13:12:03.189968      haplo_chr10_6157266_6157266     bam readcount < fastq readcount, alignment sanity check failed!

************************************************************
Can you say to me how bamsurgeon handles STAR with multiple threads? Because normally STAR puts the whole index in the memory (32GB) for one mapping.  

Thanks

**Comment by adamewing** [comment]
Hi, I just made a few changes to how STAR works that I hope will help. Someone suggested I add the --genomeLoad LoadAndKeep option to prevent re-loading the large index into memory for every mutation and I softened the requirement that the # reads in the fastq > # reads in the BAM, because I think STAR doesn't report unmapped reads in the BAM. Pull the latest and let me know if you have better luck.

--Adam
